src/main/scala/com/advancedtelematic/campaigner/http/GroupUpdateResolver.scala

```diff
@@ -0,0 +1,33 @@
+package com.advancedtelematic.campaigner.http
+
+import akka.http.scaladsl.util.FastFuture
+import com.advancedtelematic.campaigner.client.{DeviceRegistryClient, Resolver}
+import com.advancedtelematic.campaigner.data.DataType.{GroupId, Update}
+import com.advancedtelematic.campaigner.db.UpdateSupport
+import com.advancedtelematic.libats.data.DataType.Namespace
+import com.advancedtelematic.libats.messaging_datatype.DataType.DeviceId
+import slick.jdbc.MySQLProfile.api._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GroupUpdateResolver(deviceRegistry: DeviceRegistryClient, resolver: Resolver)
+                         (implicit db: Database, ec: ExecutionContext) extends UpdateSupport  {
+
+  def groupUpdates(ns: Namespace, groupId: GroupId, maxRequests: Int = 50): Future[Seq[Update]] = {
+    def fetchDevices(offset: Long, limit: Long, requests: Int = 1): Future[Seq[DeviceId]] = {
+      deviceRegistry.devicesInGroup(ns, groupId, offset, limit).flatMap { devicesInBatch =>
+        if (devicesInBatch.isEmpty || requests >= maxRequests)
```

Comment:
Not sure it is correct behaviour. IMO if we have not managed to fetch all devices it should fail.